src/AISmart.GAgent.Core/GAgentBase.cs

```diff
@@ -260,14 +160,25 @@ private async Task BaseOnActivateAsync(CancellationToken cancellationToken)
 
         // Register to itself.
         var agentGuid = this.GetPrimaryKey();
-        var streamId = StreamId.Create(CommonConstants.StreamNamespace, agentGuid);
-        var stream = StreamProvider.GetStream<EventWrapperBase>(streamId);
+        var streamIdOfThisGAgent = StreamId.Create(CommonConstants.StreamNamespace, agentGuid);
+        var streamOfThisGAgent = StreamProvider.GetStream<EventWrapperBase>(streamIdOfThisGAgent);
         foreach (var observer in Observers.Keys)
         {
-            await stream.SubscribeAsync(observer);
+            await streamOfThisGAgent.SubscribeAsync(observer);
```

Comment:
This will subscribe the agent multiple times upon deactivating then reactivating